From: Jan Grulich <jgrulich@redhat.com>
Date: Thu, 27 Jul 2023 09:49:00 +0000
Subject: Bug 1841851 - WebRTC backport: PipeWire capturer: increase buffer
 size to avoid buffer overflow r=jib,webrtc-reviewers,mjf

This is a simple backport of an WebRTC upstream change.

Upstream commit: 8fcc6df79daf1810cd4ecdb8d2ef1d361abfdc9c

Differential Revision: https://phabricator.services.mozilla.com/D183355
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/263be02fdeb0c3556e1da296eb18eb3d154eb99e
---
 .../linux/wayland/shared_screencast_stream.cc               | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc b/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
index 5b4e23c32d..0c26e7a7d5 100644
--- a/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
+++ b/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc
@@ -281,7 +281,7 @@ void SharedScreenCastStreamPrivate::OnStreamParamChanged(
 
   that->stream_size_ = DesktopSize(width, height);
 
-  uint8_t buffer[1024] = {};
+  uint8_t buffer[2048] = {};
   auto builder = spa_pod_builder{buffer, sizeof(buffer)};
 
   // Setup buffers and meta header for new format.
@@ -364,7 +364,7 @@ void SharedScreenCastStreamPrivate::OnRenegotiateFormat(void* data, uint64_t) {
   {
     PipeWireThreadLoopLock thread_loop_lock(that->pw_main_loop_);
 
-    uint8_t buffer[2048] = {};
+    uint8_t buffer[4096] = {};
 
     spa_pod_builder builder = spa_pod_builder{buffer, sizeof(buffer)};
 
@@ -482,7 +482,7 @@ bool SharedScreenCastStreamPrivate::StartScreenCastStream(
 
     pw_stream_add_listener(pw_stream_, &spa_stream_listener_,
                            &pw_stream_events_, this);
-    uint8_t buffer[2048] = {};
+    uint8_t buffer[4096] = {};
 
     spa_pod_builder builder = spa_pod_builder{buffer, sizeof(buffer)};
 
-- 
2.37.3

